contracts/nft/contracts/evm/UniversalNFT.sol

```diff
@@ -140,7 +139,7 @@ contract UniversalNFT is
         if (gasAmount > 0) {
             if (sender == address(0)) revert InvalidAddress();
             (bool success, ) = payable(sender).call{value: gasAmount}("");
-            if (!success) revert GasTokenTransferFailed();
+            if (!success) emit RefundFailed(sender, gasAmount);
```

Comment:
Right now tokens just remain in the contract. I suppose we can implement a recover function so that the owner can recover these tokens.

```solidity
  function recover(address tokenAddress, uint256 tokenAmount) public onlyOwner {
    if (tokenAddress == address(0)) {
      payable(owner()).transfer(tokenAmount);
    } else {
      IERC20(tokenAddress).transfer(owner(), tokenAmount);
    }
```

@0xM3R